### PR TITLE
Feat: Refactor settings to individual files

### LINKED ISSUE #2153: [Task]: Refactor settings to individual files
### Task description

TBD

### Requirements

TBD

### Acceptance criteria

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [X] I have added the most appropriate labels
- [X] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by begonaalvarezd**
# Description of change

This PR aims to refactor the settings routes so that each individual feature is in a standalone file.
Additionally, prop `locale` has been replaced by `localize`

## Links to any relevant issues

Closes #2153

## Type of change

- Update (a change which updates existing functionality)

## How the change has been tested

Nothing should break, this doesnt include any new feature, is just a refactor

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that my latest changes pass CI tests
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/routes/dashboard/settings/views/help/Help.svelte [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    import { Discord, Documentation, FAQ, ReportAnIssue } from './'
+
+    const settings: {
+        component: unknown
+        childRoute: HelpAndInfo
+    }[] = [
+        { component: Documentation, childRoute: HelpAndInfo.Documentation },
+        { component: FAQ, childRoute: HelpAndInfo.FAQ },
+        { component: Discord, childRoute: HelpAndInfo.Discord },
+        { component: ReportAnIssue, childRoute: HelpAndInfo.ReportAnIssue },
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/help/Help.svelte:13]: Each of these components are exactly the same maybe we can make 1 component and pass the data by props.

File: packages/shared/routes/dashboard/settings/views/security/AppLock.svelte [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        label: assignTimeoutOptionLabel(time),
+    }))
+</script>
+
+<Text type="h4" classes="mb-3">{localize('views.settings.appLock.title')}</Text>
+<Text type="p" secondary classes="mb-5">{localize('views.settings.appLock.description')}</Text>
+<Dropdown
+    onSelect={(option) => {
+        updateProfile('settings.lockScreenTimeout', option.value)
+    }}
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/security/AppLock.svelte:25]: Should we move this handler to the script block?

File: packages/shared/routes/dashboard/settings/views/security/AppLock.svelte [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<script lang="typescript">
+    import { Dropdown, Text } from 'shared/components'
+    import { localize } from 'shared/lib/i18n'
+    import { activeProfile, updateProfile } from 'shared/lib/profile'
+
+    function assignTimeoutOptionLabel(timeInMinutes) {
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/security/AppLock.svelte:6]: We need to decide on a convention for functions as we have a mixture of regular functions and arrow functions.

File: packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<script lang="typescript">
+    import { Button, Text } from 'shared/components'
+    import { localize } from 'shared/lib/i18n'
+    import { openPopup } from 'shared/lib/popup'
+
+    function handleDeleteClick() {
+        openPopup({ type: 'deleteProfile' })
+    }
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte:8]: Missing the check for non empty profiles:
```suggestion
    function handleDeleteClick() {
        if (get(get(wallet).balanceOverview).balanceRaw > 0) {
            return showAppNotification({
                type: 'error',
                message: locale('error.profile.delete.nonEmptyAccounts'),
            })
        }
        openPopup({ type: 'deleteProfile' })
    }
```

File: packages/shared/routes/dashboard/settings/views/security/Security.svelte [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                        },
+                    })
+                } else {
+                    callback(true)
+                }
+            })
+            .catch((err) => {
+                callback(false, err.error)
+            })
+    }
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/security/Security.svelte:47]: Is there a reason we export the back up when the password changes, just to spoon feed the user? If so and we use it in multiple places I would move this function out of the view component and into a library.

File: packages/shared/routes/dashboard/settings/views/advanced/index.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export { default as BalanceFinder } from './BalanceFinder.svelte'
+export { default as DeepLinks } from './DeepLinks.svelte'
+export { default as DeveloperMode } from './DeveloperMode.svelte'
```
Comment (nicole-obrien) [inline packages/shared/routes/dashboard/settings/views/advanced/index.js:3]: Not used?

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Some small changes needed / suggested

File: packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte (hunk 4 above)
Comment (nicole-obrien) [reply packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte:8]: Might need the imports too for this suggestion to work

File: packages/shared/lib/typings/routes.ts [hunk 7]
```diff
@@ -112,16 +109,21 @@ export enum AdvancedSettings {
...
 export enum HelpAndInfo {
-    // Troubleshoot = 'troubleshoot',
     Documentation = 'documentation',
     FAQ = 'faq',
     Discord = 'discord',
     ReportAnIssue = 'reportAnIssue',
 }
+
+export enum ExternalRoute {
+    Documentation = 'https://chrysalis.docs.iota.org/',
```
[the remark this answers is not in the document]
Comment (begonaalvarezd) [reply packages/shared/lib/typings/routes.ts]: updated links and references to them

File: packages/shared/routes/dashboard/settings/views/help/Help.svelte (hunk 1 above)
Comment (begonaalvarezd) [reply packages/shared/routes/dashboard/settings/views/help/Help.svelte]: refactored to single component

File: packages/shared/routes/dashboard/settings/views/security/AppLock.svelte (hunk 2 above)
Comment (begonaalvarezd) [reply packages/shared/routes/dashboard/settings/views/security/AppLock.svelte]: done

File: packages/shared/routes/dashboard/settings/views/security/AppLock.svelte (hunk 3 above)
Comment (begonaalvarezd) [reply packages/shared/routes/dashboard/settings/views/security/AppLock.svelte:10]: used function only in this file

File: packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte (hunk 4 above)
Comment (begonaalvarezd) [reply packages/shared/routes/dashboard/settings/views/security/DeleteProfile.svelte]: updated

File: packages/shared/routes/dashboard/settings/views/advanced/index.js (hunk 6 above)
Comment (begonaalvarezd) [reply packages/shared/routes/dashboard/settings/views/advanced/index.js]: removed

**Review by nicole-obrien — APPROVED** [review]
Looks ready to merge!

**Pull request merged**